File: my_cpp_pkg/src/hw_status_publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/int64.hpp"
#include "my_robot_interfaces/msg/hardware_status.hpp"

class HardwareStatusPublisherNode: public rclcpp::Node
{
    public:
        HardwareStatusPublisherNode(): Node("hardware_status_publisher")
        {
            publisher = this->create_publisher<my_robot_interfaces::msg::HardwareStatus>("hardware_status", 10);
            timer = this->create_wall_timer(
                std::chrono::milliseconds(1000),
                std::bind(&HardwareStatusPublisherNode::publichHardwareStatus, this)
            );
            RCLCPP_INFO(this->get_logger(), "HardwareStatusPublisherNode has been started");
        }

    private:
        void publichHardwareStatus()
        {
            auto msg = my_robot_interfaces::msg::HardwareStatus();
            msg.temperature = 14;
            msg.are_motors_ready = true;
            msg.degug_message = "C++ publisher is here";
            publisher->publish(msg);
        }

        rclcpp::Publisher<my_robot_interfaces::msg::HardwareStatus>::SharedPtr publisher;
        rclcpp::TimerBase::SharedPtr timer;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    //auto node = std::make_shared<rclcpp::Node>("cpp_test");
    //RCLCPP_INFO(node->get_logger(), "Hello to my Cpp Node");
    auto node = std::make_shared<HardwareStatusPublisherNode>(); 
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: my_py_pkg/my_py_pkg/hw_status_publisher.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
#from example_interfaces.msg import String
from std_msgs.msg import Int64
from my_robot_interfaces.msg import HardwareStatus

class HardwareStatusPublisherNode(Node):
    def __init__(self):
        super().__init__("hardware_status_publisher")
        self.publisher = self.create_publisher(HardwareStatus, "hardware_status", 10)
        self.timer = self.create_timer(1.0, self.publish_status)
        self.get_logger().info("Hardware Status Publisher has been started")

    def publish_status(self):
        msg = HardwareStatus()
        msg.temperature = 54
        msg.are_motors_ready = True
        msg.degug_message = "Python publisher is here"
        self.publisher.publish(msg)

def main(args = None):
    rclpy.init(args = args)
    node = HardwareStatusPublisherNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()